pymeasure/instruments/advantest/advantestR624X.py

```diff
@@ -0,0 +1,2101 @@
+            'tot', 'sct', 'osig', 'dios', 'dioe', 'ian', 'tlnk', 'wait', 'sav', 'rcl', '*sre',
+            '*ese', '*cls', 'coe', 'doe']
+
+        if not self.store_to_sequence:
+            raise ValueError("init_sequence() should be called first")
+
+        for s in valid_commands:
+            if s == command.lower()[:len(s)]:
+                self.sequence.append(command)
+
+    def init_sequence(self):
+        """ This function starts the redirection of :meth:`~.write`
+        to :meth:`~.store_sequence_command` to automatically create a sequence program.
+        """
+        self.sequence = []
+        self.store_to_sequence = True
+        self.sequence_line_count = 0
+
+    def start_sequence(self, repeat=1):
+        """ This function starts the sequence program which is
+        initiated by :meth:`~.init_sequence` and ended by :meth:`~.end_sequence`.
+        """
+        self.start_sequence_program(1, self.sequence_line_count, repeat)
+
+    def end_sequence(self):
+        """ This function ends the sequence program which is
+        initiated by :meth:`~.init_sequence`.
+        """
+        command = ''
+        self.store_to_sequence = False
+        for s in self.sequence:
+
+            # Sequence memory has a maximum of 128x100 characters
+            if len(command) + len(s) + 1 < 128:
+                command += s + ';'
+            else:
+                self.sequence_line_count += 1
+                if self.sequence_line_count > 100:
+                    raise OSError(
+                        f"{self.name} Error out of sequence memory")
+
+                self.store_sequence_command(self.sequence_line_count, command)
+                command = s + ';'
+
+        self.sequence_line_count += 1
+        self.store_sequence_command(self.sequence_line_count, command)
+
+        self.sequence = []
+
+    def sequence_wait(self, wait_mode, wait_value):
+        """ Waits for program execution and is used only for sequence programs (``WAIT``).
+
+        :param int wait_mode: Whether wait time (1) or trigger input count (2) is specified
+        :param float wait_value: Wait time or trigger input count as specified by wait_mode
+
+        This command has the following functions:
+
+          * Make the execution of the next program wait for the specified time.
+          * Makes the next program execution wait until the specified number of triggers is input.
+
+        Regardless of the wait mode, if the wait data is 0, the wait operation is not performed.
+        When the wait mode is "2", the following commands and signals can be used as trigger inputs:
+
+          * XE (XE 0, XE 1, XE 2)
+          * \\*TRG
+          * GET command (group execute trigger)
+          * Trigger input signal on rear panel
+
+        """
+        wait_mode = truncated_range(wait_mode, [1, 2])
+        self.write(f'wait {wait_mode},{wait_value}')
```

Comment:
### A suggestion / idea:

Why don't you make two different input parameters:
- wait_time
- wait_count
(Another option would be to split the method in two.)

The implementation does not have to follow the communication standard of the device, but might be more user friendly. That way, the user does not have to remember the mode values.

And you raise an error if both are selected?

Something along those lines:
```suggestion
    def sequence_wait(self, wait_time=None, wait_count=None):
        if wait_time and wait_count is None:
            self.write(f'wait 1,{wait_time}')
        elif wait_time is None and wait_count:
            self.write(f'wait 2,{wait_count}')
        else:
            raise ValueError("You have to specify the time or the count, not both")
```

Reply:
Thank you for this. When I first started porting this device I had splitted functions. I now believe it's good practice to match the functions to the GPIB commands, this makes matching the documentation the easiest.